### ISSUE TITLE
Wrong loss function

### ISSUE THREAD

**Issue by osainz59** [kazuto1011/svm-pytorch#1]
The `C` parameter of the LinearSVC should be multiplying the hinge loss and not the regularization method. Also, the l2 regularization method is defined as `w.t·w/2`, which is not equal to `torch.mean(w ** 2)`.

https://github.com/kazuto1011/svm-pytorch/blob/970eec1e1504471b547173165aefcb6003f6a9b0/main.py#L51-L52

Source: https://www.csie.ntu.edu.tw/~cjlin/papers/libsvm.pdf

**Comment by kazuto1011** [comment]
Thank you for reporting. The definition of the L2 penalty has been fixed. The `C` parameter still follows the [`sklearn.svm.SVC`](https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html) format.
